projects/hostap/libfuzzer_entry.c

```diff
@@ -0,0 +1,34 @@
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#ifndef EXTRA_ARGS
+#define EXTRA_ARGS
+#endif
+
+int fuzzer_main(int argc, char *argv[]);
+
+// Wrapper for AFL compatible fuzzer.
+//
+// TODO: Add native libFuzzer entry to project's fuzzer, as this approach has
+// noticable performance implications.
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+	int fd;
+	char input[] = "/tmp/libFuzzer-XXXXXX";
+	char* argv[] = {"fuzzer", EXTRA_ARGS input};
+
+	fd = mkstemp(input);
```

Comment:
Can you avoid disk IO please? Maybe via `fmemopen` or something similar? Disk IO inside a fuzz target may be destructive and will definitely hurt its performance.

Reply:
fmemopen doesn't work, since it would imply that I have control over the code that opens the file. In this case I unfortunately don't. I could patch the hostap repo, but I think that's more complex.

As a compromise, I've changed it to instead write the intermediate files to /dev/shm, which is guaranteed to be a ramdisk.

Reply:
Thanks for the fixes!

1) I'm not sure if `/dev/shm` is accessible in our bot environment, need to check with @oliverchang 

2) If you can't avoid creating a file, could you please use a constant name for it? Otherwise, I may blow the file system and break the bot / dismiss any good fuzzing progress

Reply:
1. /dev/shm should be available if it is identical to the docker image I tested with; AFAIK it is required to be available to be a sane environment for glibc -- http://man7.org/linux/man-pages/man7/shm_overview.7.html

2. From what I see, every docker container gets its own instance of a ramdisk mounted in the container's /dev/shm, and should not interfere with the outside world (also each ramdisk has a fixed size). One problem with making the name constant is that it's not possible to start multiple fuzzer tasks in the same container -- if this is not a problem, I'm fine with your suggestion.

Note that, if the process does not crash, the temporary file is removed again! The number of accumulated files over some period of time should only match the number of program crashes (i.e. the bugs found). After the container is destroyed, the ramdisk disappears, and so do the files.

Let me know what you think. Thanks!

Reply:
Actual fuzzing environment is documented at https://github.com/google/oss-fuzz/blob/master/docs/fuzzer_environment.md

It says "`/dev` is also unavailable.", apparently.

Regarding the temp files, I agree with you, but I remember we had some issues with other fuzz targets that attempted to do that, so I'd just follow a safer approach if possible.

Reply:
Thanks for the link. In that case, is it possible to clarify:
1. Is /tmp a ramdisk in the actual environment? In that case, using /tmp is fine I think.
2. Can you clarify if /dev/shm is really not accessible? AFAIK, it should be available otherwise other libraries may break.

Reply:
1. /tmp isn't guaranteed to be a ramdisk. 
2. /dev/shm isn't available the moment unfortunately, but may be in the near future because of changes in the pipeline.

Reply:
I found another way to avoid temporary files. PTAL.